monai/apps/pathology/handlers.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+from typing import TYPE_CHECKING, Dict, Optional
+
+import numpy as np
+
+from monai.utils import exact_version, optional_import
+
+Events, _ = optional_import("ignite.engine", "0.4.4", exact_version, "Events")
+if TYPE_CHECKING:
+    from ignite.engine import Engine
+else:
+    Engine, _ = optional_import("ignite.engine", "0.4.4", exact_version, "Engine")
+
+
+class ProbMapGenerator:
+    """
+    Event handler triggered on completing every iteration to save the probability map
+    """
+
+    def __init__(
+        self,
+        output_dir: str = "./",
+        name: Optional[str] = None,
+    ) -> None:
+        """
+        Args:
+            output_dir: output directory to save probability maps.
+            name: identifier of logging.logger to use, defaulting to `engine.logger`.
+
+        """
+        self.logger = logging.getLogger(name)
+        self._name = name
+        self.output_dir = output_dir
+        self.prob_map: Dict[str, np.ndarray] = {}
+        self.level: Dict[str, int] = {}
+        self.counter: Dict[str, int] = {}
+        self.num_done_images: int = 0
+        self.num_images: int = 0
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+
+        self.num_images = len(engine.data_loader.dataset.data_list)
```

Comment:
would `len(engine.data_loader.dataset)` work?

Reply:
ok then I think the dataset should provide an interface such as `get_num_images` otherwise this line is coupled with the dataset impl. (btw is `images` the same as `samples`?)